bftengine/src/bftengine/ReplicaImp.cpp

```diff
@@ -4054,6 +4054,7 @@ void ReplicaImp::executeRequestsAndSendResponses(PrePrepareMsg *ppMsg,
       continue;
     }
     ClientRequestMsg req((ClientRequestMsgHeader *)requestBody);
+    if (req.requestLength() == 0) continue;
```

Comment:
We need to call to clientsManager->removePendingForExecutionRequest() here, otherwise, the problem will not be resolved.

Reply:
we call it on the commit itself (before the execution)

Reply:
I see that in tryToRemovePendingRequestsForSeqNum we just mark them as committed. The call to removePendingForExecutionRequest is only one.

Reply:
right. 
Done.